### DISCUSSION TITLE
It doesn't work on some videos

### DISCUSSION THREAD

**Discussion by TheHeretic78** [aclap-dev/video-downloadhelper#1096] [Help and Q&A]
### Your question

i have a license. there's a youtube video i'm trying to download. i'm playing it, but your product doesn't detect a video. yet on other videos, it does. just wondering why.

the video does say it's an age restricted video. is that the cause? if so, can it be circumvented?

i can see the video, i'm watching it, but downloadhelper doesn't recognise a video is playing so i cannot download it.

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-GB
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "used_history_button": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 6
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

This has been reported before.  I'm not sure whether @paulrouget Paul has figured this one out yet.  But what you describe is what I have also observed, & what a number of our fellow users have observed.  There is something about age-restricted content on YouTube that confuses VDH.

Being able to watch something & being able to download it are 2 independent concepts.  For example, if something is protected by DRM, you can play it in your browser but you can't download it.  That's what DRM is for: preventing copying of copyrighted material.  I bring that up just as an analogy.  In general, the age-restricted content on YouTube is not protected by DRM.  But it suffers the same fate, that you can play it but not download it.  In the case of DRM, there is no hope VDH will ever download the content.  In the case of simply age-restricted content on YouTube, I still hold out hope Paul will figure it out.

**Comment by paulrouget** [comment]
Yes. We know of this issue. I'm working on it :)

**Reply by azizloj** [reply]
Man, so many problems lately! :-) I see that many sites no longer allow downloads, and it seems like everyone is starting to use DRM with their videos. I even failed to download videos from a small company (totalism.com). I hope you can keep up with all this pressure! :-)

**Comment by BoomerMan54** [comment]
If you're not logged in to Utube it generally won't allow access to age related videos.  

Could that be the problem?

**Comment by ghost** [comment]
Hi there! As I'm having the same problem within age restricted YT clips not to be found by vdh, I'm keeping my fingers crossed for Paul and that he'll find a solution.
With regards from Germany

**Comment by rudolfrumpel** [comment]
Hello, I have the same problem, when I didn't have a license, all videos were downloaded. But always with the ugly QR code on it. Since I bought the license, age-restricted videos have been completely ignored.
